week-four/app/src/test/java/pro/marvinhosea/movielist/ui/movies/MovieViewModelTest.kt

```diff
@@ -0,0 +1,37 @@
+package pro.marvinhosea.movielist.ui.movies
+
+import kotlinx.coroutines.GlobalScope
+import kotlinx.coroutines.launch
+import org.junit.Test
+
+import org.junit.Assert.*
+import pro.marvinhosea.movielist.data.models.Movie
+import pro.marvinhosea.movielist.data.models.POPULAR_CATEGORY
+
+class MovieViewModelTest {
+    private val modeMovieViewModel by lazy { MovieViewModel() }
+
+    @Test
+    fun testAllMoviesHasObservers() {
+        GlobalScope.launch {
```

Comment:
As mentioned, avoid global scope.

We gave links as to how to implement coroutiens in tests, and there is a lot of info online how to do it. These tests essentially are very flaky as they can both be positive and negative, or don't do anything, as coroutines don't work.

Also, from what I see underneath, your tests don't do anything. You neither have mocks, nor are doing any setup with mocks, nor are verifying behavior (all of which was shown in the video course).

I am really concerned as to what these tests even do, as you don't have any constructor parameters, are creating everything in the class itself, and who knows what kind of behavior might come out of it.